### PR TITLE
SWC-1585

### PR DESCRIPTION
**Pull request by toddis**
People Search Place. UserGroupListWidget.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/view/ComingSoonViewImpl.java [hunk 1]
```diff
@@ -47,27 +59,57 @@
...
 		this.synapseJSNIUtils = synapseJSNIUtils;
+		this.synapseClient = synapseClient;
+		this.nodeModelCreator = nodeModelCreator;
 		this.jiraErrorHelper = jiraErrorHelper;
 		this.provenanceWidget = provenanceWidget;
 		this.authenticationController = authenticationController;
 		headerWidget.configure(false);
 		header.add(headerWidget.asWidget());
-		footer.add(footerWidget.asWidget());		
+		footer.add(footerWidget.asWidget());
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/view/ComingSoonViewImpl.java:75]: remove

File: src/main/java/org/sagebionetworks/web/client/widget/user/UserGroupListWidgetViewImpl.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+			}
+			result.addStyleName("col-sm-12 col-md-6 margin-top-15");
+			result.setHeight("120px");
+		} else {
+			if (isIndividual) {
+				UserBadge userBadge = portalGinInjector.getUserBadgeWidget();
+				userBadge.configure(ownerId);
+				result = userBadge.asWidget();
+			} else {
+				BigTeamBadge teamBadge = portalGinInjector.getBigTeamBadgeWidget();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/user/UserGroupListWidgetViewImpl.java]: make small

File: src/test/java/org/sagebionetworks/web/unitclient/presenter/PeopleSearchPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	public void testCanPublicJoin() throws RestServiceException {
+		//can public join is interpretted as false if null
+		Team team = new Team();
+		team.setCanPublicJoin(null);
+		assertFalse(TeamSearchPresenter.getCanPublicJoin(team));
+		team.setCanPublicJoin(false);
+		assertFalse(TeamSearchPresenter.getCanPublicJoin(team));
+
+		team.setCanPublicJoin(true);
+		assertTrue(TeamSearchPresenter.getCanPublicJoin(team));
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/presenter/PeopleSearchPresenterTest.java]: Todd says "look at this"

File: src/test/java/org/sagebionetworks/web/unitclient/widget/user/UserGroupListWidgetTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+		mockView = mock(UserGroupListWidgetView.class);
+		mockGlobalApplicationState = mock(GlobalApplicationState.class);
+		widget = new UserGroupListWidget(mockView, mockGlobalApplicationState);
+	}
+	
+	@Test
+	public void testConfigure() {
+		List<UserGroupHeader> testUsers = getTestUsers();
+		widget.configure(testUsers);
+		verify(mockView).configure(anyList(), anyBoolean());
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/user/UserGroupListWidgetTest.java]: you could replace anyList() with eq(testUsers) and anyBoolean() with the expected isBig

File: src/test/java/org/sagebionetworks/web/unitclient/widget/user/UserGroupListWidgetTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+	}
+	
+	@Test
+	public void testGetUsers() {
+		List<UserGroupHeader> testUsers = getTestUsers();
+		widget.configure(testUsers);
+		assertTrue(widget.getUsers() == testUsers);
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/user/UserGroupListWidgetTest.java]: .equals

File: src/main/java/org/sagebionetworks/web/client/view/ComingSoonViewImpl.java (hunk 1 above)
Comment (toddis) [inline src/main/java/org/sagebionetworks/web/client/view/ComingSoonViewImpl.java:75]: Did it! Also removed something that I had added when I reimplemented the EntityTreeBrowser (woops).

File: src/main/java/org/sagebionetworks/web/client/DisplayUtils.java [hunk 6]
```diff
@@ -2103,7 +2118,13 @@ public static FlowPanel getMediaObject(String heading, String description, Click
 		} else {
 			//display default picture
 			String iconClass = defaultPictureSinglePerson ? "user" : "users";
-			HTML profilePicture = new HTML(DisplayUtils.getFontelloIcon(iconClass + " font-size-58 padding-2 imageButton userProfileImage lightGreyText margin-0-imp-before"));
+			HTML profilePicture;
+			if (clickHandler != null) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/DisplayUtils.java]: Most of this is duplicate.  Could you do something like this instead:
String clickableButtonCssClass = clickHandler != null ? "imageButton" : "";

**Pull request merged**